lib/charms/mysql/v0/mysql.py

```diff
@@ -404,6 +405,12 @@ def _on_set_password(self, event: ActionEvent) -> None:
 
         self.set_secret("app", secret_key, new_password)
 
+        if (
+            username == MONITORING_USERNAME
+            and len(self.model.relations.get(COS_AGENT_RELATION_NAME, [])) > 0
+        ):
+            self._mysql.restart_mysql_exporter()
```

Comment:
This should be also be added as an abstract method on the `MySQLBase` to enforce k8s implementation

Reply:
Addressed in [116d61f](https://github.com/canonical/mysql-operator/pull/285/commits/116d61fcb0bc433487fe20382c450f4ab67b2c20)